crates/cxx-qt/src/lib.rs

```diff
@@ -31,3 +31,15 @@ pub trait CxxQtType {
     /// The property changed signal must be emitted manually.
     unsafe fn rust_mut(self: core::pin::Pin<&mut Self>) -> core::pin::Pin<&mut Self::Rust>;
 }
+
+/// Types which have Locking implemented in the C++ generation
+///
+/// # Safety
+///
+/// This is a marker trait used to disable locking
```

Comment:
```suggestion
/// This is a marker trait used to disable locking.
/// 
/// By default, CxxQt will guard all access to the generated QObject with a recursive mutex.
/// For performance reasons it may be desirable to disable this behavior for certain QObjects.
/// You can do so by negative implementing this trait `unsafe impl !cxx_qt::Locking for qobject::T {}`.
///
/// However, this is unsafe, as it may lead to concurrent mutable access to the QObject from C++.
/// You are responsible for ensuring this does not happen!
```